src/Service/NotificationReceiverService.php

```diff
@@ -218,6 +218,16 @@ private function isAuthorized($isTestNotification, $requestUser, $requestPasswor
             return false;
         }
 
+        // Validate if username and password is configured
+        if ((is_null($userName) || is_null($password))) {
+            if ($isTestNotification) {
+                $message = 'Notifications username and/or password are not configured.';
+                throw new AuthenticationException($message);
+            }
+
+            return false;
+        }
+
```

Comment:
Can maybe this be used here?
https://github.com/Adyen/adyen-php-api-library/blob/develop/src/Adyen/Service/NotificationReceiver.php#L79

Reply:
My mistake, we cannot use it because of the php version limitation of the prestashop version 1.6